overlay.d/05rhcos/usr/lib/systemd/system/rhcos-fix-root.service

```diff
@@ -0,0 +1,17 @@
+# Fixes the partition table / kargs from machines upgrading from
+# older releases before Ignition LUKS was available
+[Unit]
+Description=Fix root device
+
+# Run before shutdown to ensure that we aren't running in a state
+# where the partition table on disk differs from the kernel's view
+Before=shutdown.target reboot.target halt.target
```

Comment:
Last time I tried to do this it didn't really work, see
https://github.com/ostreedev/ostree/blob/d046631a0f60c3e31b46eb967da29f7f94c5caaf/src/boot/ostree-finalize-staged.service#L37

Basically use `ExecStop=` - trying to start things at shutdown basically conflicts with the systemd model.

Reply:
Hmm; If I leave the unit as is just changing `ExecStart=` to `ExecStop=` then it's correctly moving the partition table but isn't updating the kargs (causing a boot failure).

Reply:
Try using `After=ostree-finalize-staged.service` too. That way we'll stage the kargs modification *before* the service that finalizes it. (Basically, the way this works is systemd runs the `ExecStop`s in reverse order).